src/test/java/by/yahor_kulesh/services/TicketServiceTest.java

```diff
@@ -0,0 +1,359 @@
+package by.yahor_kulesh.services;
+
+import by.yahor_kulesh.entity.TicketEntity;
+import by.yahor_kulesh.entity.UserEntity;
+import by.yahor_kulesh.model.tickets.Ticket;
+import by.yahor_kulesh.model.users.Client;
+import by.yahor_kulesh.repositories.TicketRepository;
+import by.yahor_kulesh.repositories.UserRepository;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+class TicketServiceTest {
+
+  private Ticket ticket;
+
+  @Mock private TicketRepository ticketRepository;
+
+  @Mock private UserRepository userRepository;
+
+  @InjectMocks private TicketService ticketService;
+
+  @Captor ArgumentCaptor<TicketEntity> ticketEntityCaptor;
+
+  @Captor ArgumentCaptor<UUID> uuidArgumentCaptor;
+
+  @BeforeEach
+  void setUp() {
+    MockitoAnnotations.openMocks(this);
+    ticket = new Ticket();
+  }
+
+  @Test
+  void insertOrUpdateTicket_TicketAdded_SavesWithCorrectUUID() {
+    ticketService.insertOrUpdateTicket(ticket);
+
+    Mockito.verify(ticketRepository, Mockito.times(1)).save(ticketEntityCaptor.capture());
+    Assertions.assertNotNull(
+        ticketEntityCaptor.getValue(), "Captured TicketEntity should not be null");
+    Assertions.assertEquals(
+        ticket.getId(),
+        ticketEntityCaptor.getValue().getId(),
+        "UUID should match the expected value");
+  }
+
+  @Test
+  void insertOrUpdateTicket_TicketAddedWithNullId_ThrowsException() {
+    ticket.setId(null);
+
+    Assertions.assertThrows(
+        IllegalArgumentException.class, () -> ticketService.insertOrUpdateTicket(ticket));
+    Mockito.verify(ticketRepository, Mockito.never()).save(Mockito.any());
+  }
+
+  @Test
+  void insertOrUpdateTicket_NullAsAParameter_ThrowsException() {
+    Assertions.assertThrows(
+        IllegalArgumentException.class, () -> ticketService.insertOrUpdateTicket(null));
+    Mockito.verify(ticketRepository, Mockito.never()).save(Mockito.any());
+  }
+
+  @Test
+  void insertOrUpdateTicket_RepositoryError_ThrowsException() {
+    Mockito.doThrow(new RuntimeException("Database Error"))
+        .when(ticketRepository)
+        .save(Mockito.isA(TicketEntity.class));
```

Comment:
it is better to create expected object and use it in stubbing instead of checking just type of the object:
```java
TicketEntity ticketEntity = ... ;

    Mockito.doThrow(new RuntimeException("Database Error"))
        .when(ticketRepository)
        .save(ticketEntity);
```